trade-finance-manager-api/src/v1/rest-mappings/mappings/gef-facilities/mapGefFacilityDates.test.js

```diff
@@ -0,0 +1,202 @@
+  const mockFacility = {
+    facilitySnapshot: {
+      ...MOCK_CASH_CONTINGENT_FACILITIES[0],
+      facilityStage: 'Issued',
+      hasBeenIssued: true,
+    },
+  };
+
+  const mockFacilityTfm = {
+    exposurePeriodInMonths: 12,
+  };
+
+  it('should return inclusionNoticeReceived as deal submissionDate if manualInclusionNoticeSubmissionDate is empty', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+    expect(result.inclusionNoticeReceived).toEqual(MOCK_GEF_DEAL.submissionDate);
+  });
+
+  it('should return manualInclusionNoticeSubmissionDate as deal submissionDate if manualInclusionNoticeSubmissionDate has a value', () => {
+    const mockMinDeal = {
+      ...MOCK_GEF_DEAL,
+      manualInclusionNoticeSubmissionDate: '16069006199999',
+    };
+
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, mockMinDeal);
+
+    expect(result.inclusionNoticeReceived).toEqual(mockMinDeal.manualInclusionNoticeSubmissionDate);
+  });
+
+  it('should return bankIssueNoticeReceived as submittedAsIssuedDate', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+
+    expect(result.bankIssueNoticeReceived).toEqual(mockFacility.facilitySnapshot.submittedAsIssuedDate);
+  });
+
+  it('should return coverStartDate as timestamp', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+
+    const expected = convertDateToTimestamp(mockFacility.facilitySnapshot.coverStartDate);
+
+    expect(result.coverStartDate).toEqual(expected);
+  });
+
+  it('should return mapped coverEndDate', () => {
+    const result = mapGefFacilityDates(mockFacility, mockFacilityTfm, MOCK_GEF_DEAL);
+
+    const mockCoverEndDate = new Date(mockFacility.facilitySnapshot.coverEndDate);
+
+    const expected = mapCoverEndDate(format(mockCoverEndDate, 'dd'), format(mockCoverEndDate, 'MM'), format(mockCoverEndDate, 'yyyy'), mockFacility);
+
+    expect(result.coverEndDate).toEqual(expected);
+  });
+
+  describe('facility end date feature flag not enabled', () => {
+    beforeAll(() => {
+      jest.mocked(isTfmFacilityEndDateFeatureFlagEnabled).mockReturnValue(false);
+    });
+
+    afterAll(() => {
+      jest.clearAllMocks();
+    });
+
+    it('should not return facility end date fields', () => {
+      const facilityWithEndDate = {
+        facilitySnapshot: {
+          ...mockFacility.facilitySnapshot,
+          isUsingFacilityEndDate: true,
+          facilityEndDate: '2021-12-08T00:00:00.000Z',
+          bankReviewDate: '2023-12-08T00:00:00.000Z',
+        },
+      };
+
+      const result = mapGefFacilityDates(facilityWithEndDate, mockFacilityTfm, MOCK_GEF_DEAL);
+
+      expect(result.isUsingFacilityEndDate).toBeUndefined();
+      expect(result.facilityEndDate).toBeUndefined();
+      expect(result.bankReviewDate).toBeUndefined();
+    });
+  });
+
+  describe('facility end date feature flag enabled', () => {
```

Comment:
Could you also add some tests for all values being undefined (To check the situation where a version 0 deal is submitted to TFM after the FF is removed)